python/ray/serve/metrics.py

```diff
@@ -0,0 +1,71 @@
+from ray.util import metrics
+from typing import Tuple, Optional, Dict, List
+from ray.serve import context
+
+DEPLOYMENT_TAG = "deployment"
+REPLICA_TAG = "replica"
+
+
+def _add_serve_metric_tags(tag_keys: Optional[Tuple[str]] = None):
+    """Add serve context tags to the tag_keys"""
+    if DEPLOYMENT_TAG in tag_keys:
+        raise ValueError(f"'{DEPLOYMENT_TAG}' tag is reserved for Ray Serve metrics")
+    if REPLICA_TAG in tag_keys:
+        raise ValueError(f"'{REPLICA_TAG}' tag is reserved for Ray Serve metrics")
+    # Get serve tag inserted:
+    if tag_keys:
+        tag_keys = (DEPLOYMENT_TAG, REPLICA_TAG) + tag_keys
+    else:
+        tag_keys = (DEPLOYMENT_TAG, REPLICA_TAG)
+    return tag_keys
+
+
+def _add_serve_metric_default_tags(default_tags: Dict[str, str]):
+    """Add serve context tags and values to the default_tags"""
+    if DEPLOYMENT_TAG in default_tags:
+        raise ValueError(f"'{DEPLOYMENT_TAG}' tag is reserved for Ray Serve metrics")
+    if REPLICA_TAG in default_tags:
+        raise ValueError(f"'{REPLICA_TAG}' tag is reserved for Ray Serve metrics")
+    default_tags[DEPLOYMENT_TAG] = context._INTERNAL_REPLICA_CONTEXT.deployment
+    default_tags[REPLICA_TAG] = context._INTERNAL_REPLICA_CONTEXT.replica_tag
```

Comment:
nit: use public method instead of global var
```suggestion
    replica_context = context.get_internal_replica_context()
    default_tags[DEPLOYMENT_TAG] = replica_context.deployment
    default_tags[REPLICA_TAG] = replica_context.replica_tag
```